packages/connect-node-conformance/src/client.ts

```diff
@@ -0,0 +1,48 @@
+import { parseArgs } from "node:util";
+import { readFileSync } from "node:fs";
+import {
+  ClientCompatRequest,
+  ClientCompatResponse,
+  ClientErrorResult,
+} from "./gen/connectrpc/conformance/v1alpha1/client_compat_pb.js";
+import invoke from "./invoke.js";
+import { createTransport } from "./transport.js";
+
+const { values: flags } = parseArgs({
+  options: {
+    json: {
+      type: "boolean",
+      default: false,
+    },
+  },
+});
+
+const req =
+  flags.json === true
+    ? ClientCompatRequest.fromJsonString(readFileSync(0, "utf-8"))
+    : ClientCompatRequest.fromBinary(
+        new Uint8Array(Buffer.from(readFileSync(0, "binary"), "binary"))
+      );
```

Comment:
Conformance clients will have to read multiple requests from stdin. For example from an earlier exploration:

```ts
import {ClientCompatRequest, ClientCompatResponse} from "../gen/connectrpc/conformance/v1alpha1/client_compat_pb";
import {protoDelimited} from "@bufbuild/protobuf";

export interface ClientUnderTest {
    name: string;
    version: string;
    run(input: AsyncIterable<ClientCompatRequest>): AsyncIterable<ClientCompatResponse>;
}

export function runNodeJs(client: ClientUnderTest): void {
    const args = process.argv.slice(2);
    if ((args.length === 1 && args[0] === "-v") || args[0] === "--version") {
        process.stdout.write(`${client.name} ${client.version}\n`);
        process.exit(0);
    }
    if (args.length !== 0) {
        process.stderr.write(
            `${client.name} accepts length-delimited ${ClientCompatRequest.typeName} on stdin and writes length-delimited ${ClientCompatResponse.typeName} to stdout\n`,
        );
        process.exit(1);
    }
    runClientWithStdio(client).then(() => {
        process.exit(0);
    }, reason => {
        process.stderr.write(`${client.name} failed: ${String(reason)}\n`);
        process.exit(1);
    });
}

async function runClientWithStdio(client: ClientUnderTest) {
    const input = protoDelimited.decStream(ClientCompatRequest, process.stdin);
    for await (const res of client.run(input)) {
        const bytes = protoDelimited.enc(res)
        process.stdout.write(bytes);
    }
}
```